### PR TITLE
Use Message Queue

### PR DESCRIPTION
**Pull request by gbirke**
Convert to typescript and use Message Queue

This is for https://phabricator.wikimedia.org/T295061

### CODE REVIEW

File: test/specs/TestCaseSerializer.ts [hunk 1]
```diff
@@ -44,4 +44,71 @@ describe('serializeTestCaseState', () => {
...
+            )
+        );
+    } );
+
+    it('fails when state name is not set', () => {
+        // @ts-expect-error
+        assert.throws(() => unserializeTestCaseState({}))
+    } );
+
+    it('fails when state name has unknown value not set', () => {
```
Comment (moiikana) [inline test/specs/TestCaseSerializer.ts]: remove 'not set' copy paste error

File: src/TestCaseSerializer.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const DEFAULT_DESCRIPTION = 'UNKNOWN DESCRIPTION - check serializer and d
 
 export function unserializeTestCaseState( stateObj: SerializedTestCaseState ): SerializableStates {
     if ( !stateObj.hasOwnProperty('stateName')) {
-        throw new Error("No stateName property found")
+        throw new Error(`Serialized test case state is missing require property "stateName"`)
```
Comment (moiikana) [inline src/TestCaseSerializer.ts]: require -> required

File: metadata_worker.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ import {Command} from "commander";
 import path from "path";
 import fs from "fs";
 import summarizeMetadata from "./src/MetadataSummarizer";
+import {workerData} from "worker_threads";
```
Comment (moiikana) [inline metadata_worker.ts]: check if this is an unused import

File: src/TestCaseSerializer.ts (hunk 2 above)
Comment (gbirke) [reply src/TestCaseSerializer.ts]: This line was removed in a later commit

**Pull request merged**